downloader/src/main/kotlin/Downloader.kt

```diff
@@ -194,7 +195,22 @@ class Downloader {
         }
 
         val startTime = Instant.now()
-        val workingTree = applicableVcs.download(target, outputDirectory, allowMovingRevisions)
+        val workingTree = try {
+            applicableVcs.download(target, outputDirectory, allowMovingRevisions)
+        } catch (e: DownloadException) {
+            // TODO: We should introduce something like a "strict" mode and only do these kind of fallbacks in
+            // non-strict mode.
+            val vcsUrlWithoutUserName = target.vcsProcessed.url.stripUserNameFromUrl()
+            if (vcsUrlWithoutUserName != target.vcsProcessed.url) {
+                // Try once more with any user name / password stripped from the URL.
+                log.info { "Falling back to trying to download from '$vcsUrlWithoutUserName'." }
```

Comment:
Log message should mention that credentials were removed from the URL.

Commit message: "commong"